test/correctness/simplify.cpp

```diff
@@ -932,6 +932,24 @@ void check_bounds() {
     check(min(min(x + z, y), w) - x, min(min(w, y) - x, z));
     check(min(min(y, x + z), w) - x, min(min(w, y) - x, z));
 
+    // More deeply-nested cancellations
```

Comment:
Beginning to question the value of these tests in the machine-generated-rule case - I just copy-pasted the rules and replaced "rewrite" with "check". I think what they usefully check is that there isn't an earlier rule that prevents the given rule from triggering on an expression that exactly matches the LHS. That has some value still - it has saved me from committing a rule that can't ever trigger a few times.

Reply:
I defer to your judgement there.